pkg/health/checker/checker.go

```diff
@@ -83,38 +83,87 @@ func (c consulHealthChecker) WatchNodeService(
 	}
 }
 
+func publishLatestHealth(inCh <-chan api.KVPairs, quitCh <-chan struct{}, resultCh chan []*health.Result) chan error {
+	errCh := make(chan error)
+
+	go func() {
+		var listed api.KVPairs
+		var ok bool
+		var err error
+
+		for {
```

Comment:
This came out looking a bit simpler than its cousin in rcstore. I think it's because of the use of continue and only reading from inCh in one place - this reduces mental bookkeeping overhead.

It's possible I missed something, so please review!